dkist/dataset/loader.py

```diff
@@ -14,6 +19,9 @@
     from asdf import ValidationError
 
 
+ASDF_FILENAME_PATTERN = r"^(?P<instrument>[A-Z-]+)_L1_(?P<timestamp>\d{8}T\d{6})_(?P<datasetid>[A-Z]+)(?P<suffix>_user_tools|_metadata)?.asdf$"
```

Comment:
Do our dataset ID's not have a fixed length?

Reply:
I think they have a minimum length, but if we mint enough datasets they will get longer iirc

Reply:
Maybe @fraserwatson can remember what the minimum length is we could add that as a constraint.

Reply:
You are correct, the dataset IDs are at least 5 characters but there is no upper bound on the length in theory.